contract/bridge/contracts/VaultRegistry.sol

```diff
@@ -393,5 +379,32 @@ abstract contract VaultRegistry is Initializable, ICollateral {
         // vault.toBeRedeemed = 0;
     }
 
-    uint256[45] private __gap;
+    function requireVaultExistence(uint256 _vaultBtcPublicKeyX) private {
+        require(_vaultBtcPublicKeyX != 0, "Vault does not exist");
+    }
+
+    function getVault(address _vaultId) external view override returns (uint256, uint256, uint256, uint256, uint256, uint256, uint256, uint256, uint256) {
+        Vault memory vault = vaults[_vaultId];
+
+        return (
+            vault.btcPublicKeyX,
+            vault.btcPublicKeyY,
+            vault.collateral,
+            vault.issued,
+            vault.toBeIssued,
+            vault.toBeRedeemed,
+            vault.replaceCollateral,
+            vault.toBeReplaced,
+            vault.liquidatedCollateral
+        );
+    }
+
+    function setVaultRewardAddress(address _vaultReward) external {
```

Comment:
The function will be called only once - when deploying the script. Actually, I'm gonna use the access control for it but since the bridge doesn't have owner, I used this solution.

Reply:
Isn't the vaultId the owner of the vault? That's the account locking the ONE's.

Reply:
So I assume you have it because you can't access that variable and you're copying it?  If that's the case, may i suggest just name it vaultId so reader knows it's the same intended address.

Reply:
There are other functions in this class that passes vaultId, so not totally sure why there is a setVaultReward though.

Reply:
> So I assume you have it because you can't access that variable and you're copying it? If that's the case, may i suggest just name it vaultId so reader knows it's the same intended address.

Could you clarify the question?

Reply:
> vaultId is the owner of the vault as well as the account locking the ONE's. **We use the same address.**

So my question, why don't you just name the variable vaultId, and also make it internal and set it during Vault creation.

Reply:
If the design calls for two different addresses, then that's different. Even if that is the case, I expect the vault owner can set this address...

Reply:
Let me clarify it more. vaultId is the vault address and msg.sender is set to vaultId when registering the vault.

Reply:
I'm aware of that. Why can't the vault address collect rewards?

Reply:
eg.

```
    function withdrawCollateral(uint256 amount) external {
        Vault storage vault = vaults[msg.sender];
```

Reply:
Ok sorry for the confusion.  I checkout the PR locally and better look at the design.  

Vault -> linked library -> VaultReward (shared across all vaults)

VaultReward init with VaultReserve

Then I think it's clearer and more idiomatic to use strong type everywhere so we know it's a contract, not a wallet.

```
function setVaultRewardAddress(IVaultReward _vaultReward)  
```

Reply:
Good